### PR TITLE
Add Atlas app-configuration alerts

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/31147

New alerts to check if some Atlas apps have unexpected configmaps/secrets

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/app-configuration.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    rules:
+    # Coming from https://gigantic.slack.com/archives/C07A03AN9JM
+    # This alert ensures our app has no unexpected configmaps.
+    - alert: ConfigmapUnexpected
+      annotations:
+        description: '{{`{{ $labels.configmap }} configmap is not expected.`}}'
+        opsrecipe: atlas-app-configuration/
+      expr: |
+        kube_configmap_info{cluster_type="management_cluster", configmap=~"loki-user-values|mimir-user-values"} > 0
+      for: 12h
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/app-configuration.yml:20]: Maybe we make this 2 days?

**Review by QuentinBisson — APPROVED** [review]

**Review by TheoBrigitte — COMMENTED** [review]
This feels strange to me. We should be in-control of what is being deployed. Having this alert rings a bell that something is not right in our deployment flow.
We should rather focus on what's causing those issues.

**Comment by QuentinBisson** [comment]
We should but it often happen that we set user values instead of using a custom branch of the shared configs for instance for Loki or Mimir so we have a few leftovers that we never cleaned up and that's causing issues when we change something and WE don't know why it has no effects

**Comment by marieroque** [comment]
@giantswarm/team-atlas any opinion on those new alerts ?

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/app-configuration.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - name: atlas-app-configuration
+    rules:
+    # Coming from https://gigantic.slack.com/archives/C07A03AN9JM
+    # This alert ensures our app has no unexpected configmaps.
+    - alert: ConfigmapUnexpected
+      annotations:
+        description: '{{`{{ $labels.configmap }} configmap is not expected.`}}'
+        opsrecipe: atlas-app-configuration/
+      expr: |
+        kube_configmap_info{cluster_type="management_cluster", configmap=~"loki-user-values|mimir-user-values"} > 0
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/app-configuration.yml:19]: - refactoring to shorten CM name when we add more app names
- don't we have some user-values named `[clusternaem]-[app]-user-values`?
```suggestion
        kube_configmap_info{cluster_type="management_cluster", configmap=~".*(loki|mimir)-user-values"} > 0
```

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**